### ISSUE TITLE
Key-spacing reset after nested

### ISSUE THREAD

**Issue by gjuchault** [eslint/eslint#4883]
```js
const a = {
    aaaa: 'a',
    b   : 'b',
    v   : null,
    x   : 'c',
    q   : [
        'ok'
    ],
    p   : 50,
    l   : 'foo'
};

a.type = 'x';
```

With `"key-spacing": [2, { "align": "colon" }]`

Outputs :
```
   9:5  error  Extra space after key "p"  key-spacing
  10:5  error  Extra space after key "l"  key-spacing
```

Actually, this is aligned.

Version : eslint v1.10.3

**Comment by eslintbot** [comment]
@gjuchault Thanks for the issue! If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

**Comment by btmills** [comment]
This is a duplicate of #2668, which is accepted. @gjuchault want to send a PR for that issue?
